ui/src/session.ts

```diff
@@ -126,9 +129,87 @@ export const updateCoCo = (coco: CoCo): void =>
     settings: { ...get(settings), coco },
   });
 
-export const parseSeedsInput = (input: string): string[] => {
-  return input
-    .replace(/\r\n|\n|\r|\s/gm, ",")
-    .split(",")
-    .filter(seed => seed !== "");
+const VALID_SEED_MATCH = /^[\w\d]{54}@[\w\d.]*:[\d]{1,5}$/;
+
+const checkSeedUniqueness = (seed: string): Promise<boolean> => {
+  return Promise.resolve(!get(settings).coco.seeds.includes(seed));
+};
+
+export const seedValidation = createValidationStore(
+  {
+    format: {
+      pattern: VALID_SEED_MATCH,
+      message: "This is not a valid seed address",
+    },
+  },
+  [
+    {
+      promise: checkSeedUniqueness,
+      validationMessage: "This seed already exists",
+    },
+  ]
+);
+
+let restartNotificationVisible = false;
+
+const reloadProxy = async () => {
+  await window.fetch("http://localhost:8080/v1/control/reload");
+  notification.info("Reloading peer configuration, hang in there…");
+
+  // Prevent the user from interacting with the UI while proxy is reloading.
+  screen.lock();
+  let proxyStopped = true;
+
+  while (proxyStopped) {
+    await new Promise(resolve => setTimeout(resolve, 1000));
+    try {
+      const resp = await api.get<Session>(`session`);
+      if (resp) {
+        proxyStopped = false;
+      }
+    } catch {
+      // Ignore failed attempts to connect.
+    }
+  }
+  screen.unlock();
+
+  restartNotificationVisible = false;
+};
+
+const showRestartNotification = (): void => {
+  // The restart notification is sticky and will stay visible until the user
+  // clicks the "Restart" button, no need to show it multiple times.
+  if (restartNotificationVisible) return;
+
+  notification.error(
```

Comment:
I understand that we want to make sure the user doesn't miss this, but isn't it a bit of an abuse to show an error here? Can't we have something like a warning variant?

Reply:
Good point. In the [style guide](https://www.figma.com/file/owmgsbs6lnUt8R1bixstCA/%F0%9F%8C%B1%F0%9F%86%99-Radicle-Upstream?node-id=586%3A0) we only have 2 levels so far.

Was actually thinking of renaming `error` in the code to `attention`.

Reply:
Maybe @brandonhaslegs can let us know if we need a third variant?

Reply:
If we had persistent notifications (notifications that don't go away until you interact with them) then I would say keeping it black is fine. But this notification is **critical.** The user **must** restart to see results. Therefore, the user **must** see this notification. This is why I thought making it an error would be best. I don't see this as abuse, I see this as being helpful. If they miss this notification, they're confused, the app is "broken," they close the app and never come back.

Maybe I could settle for adding an "Attention" color. If we decide that's what we want to do, here's a design for it: https://www.figma.com/file/owmgsbs6lnUt8R1bixstCA/%F0%9F%8C%B1%F0%9F%86%99-Radicle-Upstream?node-id=5863%3A47629

Reply:
Thanks for the context @brandonhaslegs - if I understand the code correctly the notification sticks around with this change, but maybe I'm off.

Reply:
@xla correct, we have persistent notifications now. Will change the notification type to `info` then, which will make the sticky notification black.